### PR TITLE
change http links to https links 

### PR DESCRIPTION
**Pull request by Mrlinkwii**
## Summary

Describe your changes here.
change http to https on links 
Fixes #Issue_Number (if available)
Contributes to #4504

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/docs?pullRequest=19968) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/docs?pullRequest=19968) before we can accept your contribution.<br/><br/>:x: Mrlinkwii [sign now](https://cla.dotnetfoundation.org/dotnet/docs?pullRequest=19968)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/docs?pullRequest=19968) it.</sub>

File: docs/architecture/cloud-native/elastic-search-in-azure.md [hunk 1]
```diff
@@ -55,7 +55,7 @@ This chapter presented a detailed look at data in cloud-native systems. We start
 
 - [Exploring the different types of NoSQL Databases Part II](https://www.3pillarglobal.com/insights/exploring-the-different-types-of-nosql-databases)
 
-- [On RDBMS, NoSQL and NewSQL databases. Interview with John Ryan](http://www.odbms.org/blog/2018/03/on-rdbms-nosql-and-newsql-databases-interview-with-john-ryan/)
+- [On RDBMS, NoSQL and NewSQL databases. Interview with John Ryan](https://www.odbms.org/blog/2018/03/on-rdbms-nosql-and-newsql-databases-interview-with-john-ryan/)
```
Comment (adegeo) [inline docs/architecture/cloud-native/elastic-search-in-azure.md:58]: This link reports bad cert

File: docs/architecture/containerized-lifecycle/design-develop-containerized-apps/common-container-design-principles.md [hunk 2]
```diff
@@ -11,7 +11,7 @@ Ahead of getting into the development process there are a few basic concepts wor
...
 
-You might find a scenario in which you want multiple processes running in a single container. In any architecture document, there's never a "never," nor is there always an "always." For scenarios requiring multiple processes, a common pattern is to use [Supervisor](http://supervisord.org/).
+You might find a scenario in which you want multiple processes running in a single container. In any architecture document, there's never a "never," nor is there always an "always." For scenarios requiring multiple processes, a common pattern is to use [Supervisor](https://supervisord.org/).
```
Comment (adegeo) [inline docs/architecture/containerized-lifecycle/design-develop-containerized-apps/common-container-design-principles.md:14]: This one is just spinning for me, it won't connect.

File: docs/architecture/containerized-lifecycle/docker-devops-workflow/docker-application-outer-loop-devops-workflow.md [hunk 3]
```diff
@@ -89,7 +89,7 @@ With these Visual Studio Team Services tasks, a build Linux-Docker Host/VM provi
 >   <https://docs.microsoft.com/archive/blogs/stevelasker/building-net-core-linux-docker-images-with-visual-studio-team-services>
 >
 > - Building a Linux-based Visual Studio Team Service build machine with Docker support: \
->   <http://donovanbrown.com/post/2016/06/03/Building-a-Linux-Based-Visual-Studio-Team-Service-Build-Machine-with-Docker-Support>
+>   <https://donovanbrown.com/post/2016/06/03/Building-a-Linux-Based-Visual-Studio-Team-Service-Build-Machine-with-Docker-Support>
```
Comment (adegeo) [inline docs/architecture/containerized-lifecycle/docker-devops-workflow/docker-application-outer-loop-devops-workflow.md:92]: This link has a bad cert

File: docs/index.yml [hunk 4]
```diff
@@ -234,7 +234,7 @@ additionalContent:
             text: Game development with Visual Studio
           - url: https://docs.cryengine.com/display/CEPROG/C%23+Programming
             text: Learn how to use CRYENGINE to build games with C#
-          - url: http://www.monogame.net/documentation/?page=main
+          - url: https://www.monogame.net/documentation/?page=main
```
Comment (adegeo) [inline docs/index.yml:237]: That link forward to this one:

```suggestion
          - url: https://docs.monogame.net/?page=main
```

**Review by adegeo — CHANGES_REQUESTED** [review]

**Comment by adegeo** [comment]
Hello! Thanks for opening this PR. There are a few things to resolve. A few links don't seem to work, so I think until they are fixed, they should be reverted. Also, the monogame link forwards to another link, so I added a suggestion to accept that fixes that.

Thanks!

File: docs/index.yml (hunk 4 above)
Comment (Youssef1313) [reply docs/index.yml:237]: Is the `page=main` parameter query necessary?
I don't see a difference when I open the page with or without it.

**Review by adegeo — APPROVED** [review]
Thanks! Sorry for the delay!

File: docs/architecture/containerized-lifecycle/docker-devops-workflow/docker-application-outer-loop-devops-workflow.md [hunk 5]
```diff
@@ -89,7 +89,7 @@ With these Visual Studio Team Services tasks, a build Linux-Docker Host/VM provi
 >   <https://docs.microsoft.com/archive/blogs/stevelasker/building-net-core-linux-docker-images-with-visual-studio-team-services>
 >
 > - Building a Linux-based Visual Studio Team Service build machine with Docker support: \
->   <https://donovanbrown.com/post/2016/06/03/Building-a-Linux-Based-Visual-Studio-Team-Service-Build-Machine-with-Docker-Support>
+>   <http://donovanbrown.com/post/2016/06/03/Building-a-Linux-Based-Visual-Studio-Team-Service-Build-Machine-with-Docker-Support>
```
Comment (adegeo) [inline docs/architecture/containerized-lifecycle/docker-devops-workflow/docker-application-outer-loop-devops-workflow.md]: Looks like this link actually redirects (as http) to a good https site now: https://www.donovanbrown.com/post/Building-a-Linux-Based-Visual-Studio-Team-Service-Build-Machine-with-Docker-Support

Comment (adegeo) [reply docs/architecture/containerized-lifecycle/docker-devops-workflow/docker-application-outer-loop-devops-workflow.md]: I fixed it by editing your file 👍

**Review by adegeo — APPROVED** [review]

**Pull request merged**